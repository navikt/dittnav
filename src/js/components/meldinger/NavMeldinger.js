import React, { Component } from 'react';
import PropTypes from 'prop-types';
import conf from 'js/Config';
import i18n from 'translations/i18n';
import { FormattedMessage as F, injectIntl, intlShape } from 'react-intl';

class NavMeldinger extends Component {
  render() {
    const { navMessagesCount } = this.props;
    const { numberToWord } = i18n[this.props.intl.locale];

    if (navMessagesCount === 0) return null;
    return (
      <a className="message clickable" data-ga="Dittnav/Varsel/Nav-melding" href={`${conf.dittNav.SERVICES_URL}${conf.ARBEID_PATH}?goto=${conf.MELDINGER_NAV_PATH}`}>
        <span className="icon default-icon" aria-label="default-ikon" />
        <div className="texts">
          <p><F id={navMessagesCount !== 1 ? 'nav.melding.flere' : 'nav.melding.en'} values={{ count: numberToWord(navMessagesCount) }} /></p>
          <p><F id={navMessagesCount !== 1 ? 'melding.se.oversikt' : 'melding.se'} /></p>
        </div>
      </a>
    );
  }
}

NavMeldinger.propTypes = {
  navMessagesCount: PropTypes.number.isRequired,
  intl: intlShape.isRequired, // eslint-disable-line react/no-typos
};

export default injectIntl(NavMeldinger);
