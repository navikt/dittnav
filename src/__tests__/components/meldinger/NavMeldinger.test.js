import * as React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import NavMeldinger from 'js/components/meldinger/NavMeldinger'
import wrapIntl from 'js/IntlTestHelper';

test('NavMeldinger without messages', () => {
  const component = ReactTestRenderer.create(wrapIntl(<NavMeldinger navMessagesCount={0} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('NavMeldinger with 1 message', () => {
  const component = ReactTestRenderer.create(wrapIntl(<NavMeldinger navMessagesCount={1} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('NavMeldinger with 5 messages', () => {
  const component = ReactTestRenderer.create(wrapIntl(<NavMeldinger navMessagesCount={5} />));
  expect(component.toJSON()).toMatchSnapshot();
});
