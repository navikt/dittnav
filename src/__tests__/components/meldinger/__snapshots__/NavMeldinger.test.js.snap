// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMeldinger with 1 message 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/Nav-melding"
  href="http://localhost/sbl/nav_security_check?goto=/sbl/as/minside/meldinger/meldingerNAV.do"
>
  <span
    aria-label="default-ikon"
    className="icon default-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har fått en ny melding fra NAV
      </span>
    </p>
    <p>
      <span>
        Se melding
      </span>
    </p>
  </div>
</a>
`;

exports[`NavMeldinger with 5 messages 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/Nav-melding"
  href="http://localhost/sbl/nav_security_check?goto=/sbl/as/minside/meldinger/meldingerNAV.do"
>
  <span
    aria-label="default-ikon"
    className="icon default-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har fått fem nye meldinger fra NAV
      </span>
    </p>
    <p>
      <span>
        Oversikt over meldinger
      </span>
    </p>
  </div>
</a>
`;

exports[`NavMeldinger without messages 1`] = `null`;
